### PR TITLE
Allow referrers to view submitted application overview

### PR DESCRIPTION
**Pull request by patrickjfl**
# Context

Jira ticket: https://dsdmoj.atlassian.net/jira/software/c/projects/CAS2/boards/1347?selectedIssue=CAS2-196

This work is based on the API solution sketched out here: https://github.com/ministryofjustice/hmpps-approved-premises-api/compare/196-return-status-updates?expand=1. It should not be merged until that solution is merged and the models are updated. If a different API solution is preferred this branch will require changes.

# Changes in this PR

Creates an application overview page for referrers, to parallel the assessor view, albeit with minor modifications - the 'Update status' button is not visible. 

## Screenshots of UI changes

Screenshot 2024-02-12 at 14 12 00

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by Harriethw** [comment]
API PR was merged and should be in dev etc. soon. https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1455

**Comment by Harriethw** [comment]
the CI pipeline is currently failing so the API work is still in dev FYI https://mojdt.slack.com/archives/C03K0HB0LBE/p1707411960101449?thread_ts=1707319458.186559&cid=C03K0HB0LBE

File: server/utils/applications/utils.ts [hunk 1]
```diff
@@ -83,10 +88,15 @@ export const getSubmittedTimelineEvent = (submitterName: string, submittedAt: st
   }
 }
 
-export const getApplicationTimelineEvents = (application: Cas2SubmittedApplication): Array<UiTimelineEvent> => {
+export const getApplicationTimelineEvents = (
```
Comment (libuk) [inline server/utils/applications/utils.ts:91]: Should we add a test to cover this scenario?

File: integration_tests/tests/apply/view_application_overview_as_referrer.cy.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+//    As an referrer
+//    I want to view the overview of a particular submitted application
+//
+//  Scenario: visits submitted application overview page
+//    Given a submitted application exists
+//    And I am logged in as a referrer
+//    When I visit the application overview for that application
+//    Then I see the submitted application overview
+//
+//  Scenario: first visit to submitted application overview page
```
Comment (libuk) [inline integration_tests/tests/apply/view_application_overview_as_referrer.cy.ts:12]: I think we can remove this text as we aren't testing this scenario.

**Review by libuk — APPROVED** [review]
Nice work! Just a couple of comments.

File: integration_tests/tests/apply/view_application_overview_as_referrer.cy.ts (hunk 2 above)
Comment (patrickjfl) [reply integration_tests/tests/apply/view_application_overview_as_referrer.cy.ts:12]: oh yeah, good spot :)

**Pull request merged**